h/static/scripts/annotator/plugin/pdf-metadata.js

```diff
@@ -0,0 +1,78 @@
+'use strict';
+
+/**
+ * This PDFMetadata service provides implementations for extracting fields
+ * of a pdf document from the input PDFViewerApplication object.
+ *
+ * The fields extracted are:
+ * #1. urn - a string which uniquely identifies a pdf document.
+ * #2. title - a string representing the title of the pdf document.
+ * #3. link - an array of two objects, the urn and the url strings of the pdf.
+ * #4. documentFingerprint - the string representing the pdf fingerprint.
+ */
+function PDFMetadata(pdfViewerApplication) {
+  this._pdfViewerApplication = pdfViewerApplication;
+  this._documentLoaded = new Promise(function (resolve) {
+    var finish;
+    finish = function () {
+      window.removeEventListener('documentload', finish);
+      resolve();
+    };
+
+    if (this._pdfViewerApplication.documentFingerprint) {
+      resolve();
+    }
+
+    window.addEventListener('documentload', finish);
+  }.bind(this));
+}
+
+/**
+ * Returns a promise of a PDF URN string after the pdf loads
+ * succesfully and the 'documentload' event fires.
+ */
+PDFMetadata.prototype.getUrn = function () {
+  return this._documentLoaded.then(function () {
+    return 'urn:x-pdf:' + this._pdfViewerApplication.documentFingerprint;
+  }.bind(this));
+};
+
+/**
+ * Returns a promise of an object with the fields
+ * title(string), link(array) and documentFingerprint(string)
+ * after the pdf loads succesfully and the 'documentload' event fires.
+ */
+PDFMetadata.prototype.getMetadata = function () {
+  return this._documentLoaded.then(function () {
+    var info;
```

Comment:
Where possible it is preferable to put the declaration and initialization on one line, eg. `var info = this._pdfViewerApplication.documentInfo` instead of `var info; info = this. ...`. There are two reasons for this: Fewer lines of code, and when we convert the codebase to ES6, we can use `const varName = <initializer>` to help guard against accidental re-assignments.